### PR TITLE
docker: add Docker template for a build environment

### PR DESCRIPTION
**Pull request by chaudron**
Add a Docker template that can be used as a build environment instead
of installing all build tools locally. It also includes documentation
on how to set it up, and use it.

Signed-off-by: Eelco Chaudron <echaudron@gmail.com>

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+INFO: 74 processes: 74 processwrapper-sandbox.
+INFO: Build completed successfully, 146 total actions
+[root@ebplane ebplane]# 
+```
+
+And finish it off with running the unit tests:
+
+```
+[root@ebplane ebplane]# bazel test ...:all
+DEBUG: /root/.cache/bazel/_bazel_root/4a41d10c950280d61cfeba75245d2362/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:9: rbe_default not using checked in configs; Bazel version 0.29.1 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"]}
```
Comment (epeer-juniper) [inline docker/README.md]: Are we supposed to see this? What does it mean?

Comment (chaudron) [reply docker/README.md]: Good question, I see this in all my Bazel experiments so far. I have not enabled any debugging or other Bazel settings.

I assume you also see this when trying it on your Mac?

Comment (chaudron) [reply docker/README.md]: Good question, I see this in all my Bazel experiments so far. I have not enabled any debugging or other Bazel settings.

I assume you also see this when trying it on your Mac?

Comment (epeer-juniper) [reply docker/README.md]: Indeed, I think PR #32 fixes this though.

Comment (chaudron) [reply docker/README.md]: Ok will rerun it with that PR applied and update the documentation and this PR

**Comment by chaudron** [comment]
Updated patch to use #32 for the build output

**Comment by chaudron** [comment]
Anything blocking this getting merged?

**Pull request merged**